components/BlobToEntry.coffee

```diff
@@ -23,100 +24,132 @@ handleGraph = (sha, content, entry, project, out, callback) ->
       entry.local.endTransaction sha
       # Ensure the graph is marked as not changed since SHA
       entry.local.properties.changed = false
-      out.send entry.local
-      do callback
+      collections.addToList project.graphs, entry.local
+      callback null, entry.local
       return
 
     graph.properties.name = entry.remote.name
     graph.name = entry.remote.name
-    graph.properties.id = generateId project, entry
+    graph.properties.id = uuid.v4()
     graph.properties.environment = {} unless graph.properties.environment
     graph.properties.environment.type = project.type unless graph.properties.environment.type
-    project.graphs.push graph
-    out.send graph
+    collections.addToList project.graphs, graph
+    callback null, graph
     do callback
 
-handleComponent = (sha, content, entry, project, out, callback) ->
+handleComponent = (sha, content, entry, project, callback) ->
   if entry.local
     entry.local.code = content
     entry.local.sha = sha
     entry.local.changed = false
-    out.send entry.local
-    do callback
+    collections.addToList project.components, entry.local
+    callback null, entry.local
     return
   newEntry =
-    id: generateId project, entry
+    id: uuid.v4()
     project: project.id
     name: entry.remote.name
     code: content
     language: entry.remote.language
     sha: sha
     changed: false
-  project.components.push newEntry
-  out.send newEntry
-  do callback
+  collections.addToList project.components, newEntry
```

Comment:
What is this `colletions.addToList` thing? If it is just wrapping `list.push()` that serves no purpose and just obscures the code?

Reply:
@jonnor it either pushes, or replaces existing